Session.cc

```diff
@@ -1257,10 +1297,14 @@ void Session::SendFileEvent(int32_t file_id, CARTA::EventType event_type, uint32
 void Session::SendPendingMessages() {
     // Do not parallelize: this must be done serially
     // due to the constraints of uWS.
-    std::vector<char> msg;
+    std::pair<std::vector<char>, bool> msg;
     if (_connected) {
         while (_out_msgs.try_pop(msg)) {
-            _socket->send(msg.data(), msg.size(), uWS::BINARY);
+            if (msg.second) { // Compress the message
```

Comment:
this can be simplified to just `_socket->send(msg.first.data(), msg.first.size(), uWS::BINARY, nullptr, nullptr, msg.second);`